must-gather/collection-scripts/gather_namespace_resources

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+# Expect base collection path as an argument
+BASE_COLLECTION_PATH=$1
+
+# Use PWD as base path if no argument is passed
+if [ "${BASE_COLLECTION_PATH}" = "" ]; then
+    BASE_COLLECTION_PATH=$(pwd)
+fi
+
+# Add general resources to list if necessary
+
+# Resource List
+resources=()
+# collect storagecluster resources
+
+# TODO: Re enable the collection of storagecluster via inspect command
+# resources+=(storageclusters)
+
+# collect OB/OBC resoureces
+resources+=(objectbucketclaims)
+resources+=(objectbuckets)
+
+# collection path for OC commands
+mkdir -p ${BASE_COLLECTION_PATH}/oc_output/
+
+# Command List
+commands=()
+
+# collect oc output of OC commands
+commands+=("get clusterversion -oyaml")
+commands+=("get pods -owide -n openshift-storage")
+commands+=("get nodes --show-labels")
+commands+=("describe nodes")
+commands+=("describe pods -n openshift-storage")
+commands+=("get pvc --all-namespaces")
+commands+=("get pv")
+commands+=("get infrastructures.config -oyaml")
+commands+=("get subscription -n openshift-storage")
+commands+=("get csv -n openshift-storage")
+commands+=("get sc")
+commands+=("get installplan -n openshift-storage")
+commands+=("get events -n openshift-storage")
+commands+=("get all -n openshift-storage")
+commands+=("clusterrole -n openshift-storage")
+commands+=("clusterrolebinding -n openshift-storage")
+commands+=("role -n openshift-storage")
+commands+=("rolebinding -n openshift-storage")
+
+# collect yaml output of OC commands
+oc_yamls=()
+oc_yaml+=("pvc")
+oc_yaml+=("csv")
+oc_yaml+=("sc")
+oc_yaml+=("installplan")
+oc_yaml+=("subscription")
+oc_yaml+=("clusterrole")
+oc_yaml+=("clusterrolebinding")
+oc_yaml+=("role")
+oc_yaml+=("rolebinding")
+
+namespaces=$(oc get deploy --all-namespaces -o go-template --template='{{range .items}}{{if .metadata.labels}}{{printf "%s %v" .metadata.namespace (index .metadata.labels "olm.owner")}} {{printf "\n"}}{{end}}{{end}}' | grep ocs-operator | awk '{print $1}' | uniq)
+# Inspecting the namespace where ocs-cluster is installed
+for ns in $namespaces; do
+    echo "collecting dump of namespace" | tee -a  "${BASE_COLLECTION_PATH}"/gather-debug.log
+    oc adm --dest-dir="${BASE_COLLECTION_PATH}" inspect ns/"${ns}" >> "${BASE_COLLECTION_PATH}"/gather-debug.log 2>&1
+    echo "collecting dump of clusterresourceversion" | tee -a  "${BASE_COLLECTION_PATH}"/gather-debug.log
+    for oc_yaml in "${oc_yamls[@]}"; do
+        # shellcheck disable=SC2129
+        oc adm --dest-dir="${BASE_COLLECTION_PATH}" inspect "${oc_yaml}" -n "${ns}" >> "${BASE_COLLECTION_PATH}"/gather-debug.log 2>&1
+done
+
+
+# NOTE: This is a temporary fix for collecting the storagecluster as we are not able to collect the storagecluster using the inspect command
+{ oc get storageclusters --all-namespaces -o yaml; } > $BASE_COLLECTION_PATH/storagecluster.yaml 2>&1
```

Comment:
We can restrict it to "openshift-storage". StorageCluster can't be created in other namespaces.

Reply:
ok